Structures.cpp

```diff
@@ -81,31 +85,44 @@ double averageMark(Student student) {
     return sum / static_cast<double>((sizeof(student.marks) / sizeof(int)));
 }
 
-int compareByAverageMark(const void* a, const void* b)
+/*int compareByAverageMark(const void* a, const void* b)
 {
     if (averageMark(*(Student*)a) < averageMark(*(Student*)b)) return -1;
     if (averageMark(*(Student*)a) == averageMark(*(Student*)b)) return 0;
     if (averageMark(*(Student*)a) > averageMark(*(Student*)b)) return 1;
+}*/
+
+bool compareByAverageMark(const Student& s1, const Student& s2) {
+    return averageMark(s1) > averageMark(s2);
 }
 
 void sortByAverageMark(Student (&students)[CLASS_SIZE]) {
-    std::qsort(students, CLASS_SIZE, sizeof(Student), compareByAverageMark);
+    //std::qsort(students, CLASS_SIZE, sizeof(Student), compareByAverageMark);
+    //std::sort(students, students + CLASS_SIZE, [](const Student &s1, const Student &s2) {return averageMark(s1) > averageMark(s2);});
+    std::sort(students, students + CLASS_SIZE, compareByAverageMark);
 }
 
 Student* getBestStudentPtr(Student(&students)[CLASS_SIZE]) {
-    sortByAverageMark(students);
-    return &(students[CLASS_SIZE - 1]);
+    Student *bestStudentPtr = students;
+
+    for (int i = 1; i <= CLASS_SIZE; i++) {
+        if (compareByAverageMark(students[i], *bestStudentPtr)) {
+            bestStudentPtr = students + i;
+        }
+    }
```

Comment:
you can do lika that. but in this way you're calculating average of the best student on each iteration, but you should do it only when you change it